test/jasmine/tests/command_test.js

```diff
@@ -0,0 +1,477 @@
+                });
+
+                it('with traces specified in reverse order', function() {
+                    var result = Plots.computeAPICommandBindings(gd, 'restyle', [{'marker.size': [7, 5]}, [1, 0]]);
+                    expect(result).toEqual([{type: 'data', prop: 'marker.size', traces: [1, 0], value: [7, 5]}]);
+                });
+
+                it('with two values and a single trace specified', function() {
+                    var result = Plots.computeAPICommandBindings(gd, 'restyle', [{'marker.size': [7, 5]}, [0]]);
+                    expect(result).toEqual([{type: 'data', prop: 'marker.size', traces: [0], value: [7]}]);
+                });
+
+                it('with two values and a different trace specified', function() {
+                    var result = Plots.computeAPICommandBindings(gd, 'restyle', [{'marker.size': [7, 5]}, [1]]);
+                    expect(result).toEqual([{type: 'data', prop: 'marker.size', traces: [1], value: [7]}]);
+                });
+            });
+
+            describe('and multiple attributes', function() {
+                it('with a scalar value', function() {
+                    var result = Plots.computeAPICommandBindings(gd, 'restyle', [{'marker.size': 7, 'text.color': 'blue'}]);
+                    expect(result).toEqual([
+                        {type: 'data', prop: 'marker.size', traces: null, value: 7},
+                        {type: 'data', prop: 'text.color', traces: null, value: 'blue'}
+                    ]);
+                });
+            });
+        });
+
+        describe('with mixed notation', function() {
+            it('and nested object and nested attr', function() {
+                var result = Plots.computeAPICommandBindings(gd, 'restyle', [{
+                    y: [[3, 4, 5]],
+                    'marker.size': [10, 20, 25],
+                    'line.color': 'red',
+                    line: {
+                        width: [2, 8]
+                    }
+                }]);
+
+                // The results are definitely not completely intuitive, so this
+                // is based upon empirical results with a codepen example:
+                expect(result).toEqual([
+                    {type: 'data', prop: 'y', traces: [0], value: [[3, 4, 5]]},
+                    {type: 'data', prop: 'marker.size', traces: [0, 1], value: [10, 20]},
+                    {type: 'data', prop: 'line.color', traces: null, value: 'red'},
+                    {type: 'data', prop: 'line.width', traces: [0, 1], value: [2, 8]}
+                ]);
+            });
+
+            it('and traces specified', function() {
+                var result = Plots.computeAPICommandBindings(gd, 'restyle', [{
+                    y: [[3, 4, 5]],
+                    'marker.size': [10, 20, 25],
+                    'line.color': 'red',
+                    line: {
```

Comment:
Oh no! Good catch. I changed the order so that this is no longer accurate 😞 I need to fix this. Sigh.